server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -242,6 +245,69 @@ public CloseableIterator<DataSegmentPlus> retrieveUnusedSegmentsPlus(
     );
   }
 
+  public Set<SegmentId> retrieveUsedSegmentIds(
+      final String dataSource,
+      final Interval interval
+  )
+  {
+    return retrieveSegmentIds(dataSource, Collections.singletonList(interval));
+  }
+
+  private Set<SegmentId> retrieveSegmentIds(
```

Comment:
should this also be called retrieveUsedSegmentIds?

Reply:
Yes, it should. Thanks, I'll make the change.